SimTracker/TrackerHitAssociation/interface/trackerHitAssociationHeterogeneous.h

```diff
@@ -0,0 +1,78 @@
+#ifndef SimTracker_TrackerHitAssociation_plugins_trackerHitAssociationHeterogeneousProduct_h
+#define SimTracker_TrackerHitAssociation_plugins_trackerHitAssociationHeterogeneousProduct_h
+
+
+#include "CUDADataFormats/Common/interface/HeterogeneousSoA.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/copyAsync.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/cudaCheck.h"
+
+
+
+namespace trackerHitAssociationHeterogeneous {
+
+  struct ClusterSLView {
+    using Clus2TP = std::array<uint32_t, 7>;
+
+    Clus2TP* links_d;
+    uint32_t* tkId_d;
+    uint32_t* tkId2_d;
+    uint32_t* n1_d;
+    uint32_t* n2_d;
+
+  };
+
+  template <typename Traits>
+  class Product {
```

Comment:
In principle it would be nicer to have different naming than `trackerHitAssociationHeterogeneous::Product` (similar to data format class names in `CUDADataFormats`), but OTOH I'm not sure if it is worth of the effort.

Reply:
sure. at some point we need to consolidate it.